packages/jest-environment-puppeteer/src/PuppeteerEnvironment.js

```diff
@@ -115,16 +115,25 @@ class PuppeteerEnvironment extends NodeEnvironment {
   }
 
   async teardown() {
-    const config = await readConfig()
-    this.global.page.removeListener('pageerror', handleError)
+    const { page, context, browser, puppeteerConfig } = this.global
 
-    if (config.browserContext === 'incognito') {
-      await this.global.context.close()
+    if (page) {
+      page.removeListener('pageerror', handleError)
+    }
```

Comment:
You are checking `if (page)` on line L120 and L129. If it's supposed to be run only if `page` is `true` you can combine those two if blocks into one. Logic would stay the same.

Reply:
This should be dont always, if page is set.